src/unstract/sdk/utils/tool_utils.py

```diff
@@ -50,13 +55,17 @@ def get_hash_from_file(file_path: str) -> str:
         h = sha256()
         b = bytearray(128 * 1024)
         mv = memoryview(b)
-        with open(file_path, "rb", buffering=0) as f:
-            while n := f.readinto(mv):
-                h.update(mv[:n])
+
+        file_contents = fs.read(path=file_path, mode="rb")
+        mv = memoryview(file_contents)
+        h.update(mv[: len(file_contents)])
```

Comment:
@gaya3-zipstack This will work, but not efficient. This is one of the downsides of not exposing the file handle explicitly.

1. `buffering=0` is required to disable buffering.
2. `memoryview` (whose size is determined by underlying `bytearray`) is a temporary placeholder for chunks read from a file. Instead here we are reading the whole file in one shot for which `memoryview` intermediate step is not required at all. Also if the file size is large, this will try to keep the whole file in memory.

For now the easiest route will be to add a function `hash_file` in the underlying API interface, and move this same logic there, as in this way this function will have direct access to the file handle.

Reply:
Yeah @hari-kuriakose . I explored a bit and looks like that is the best for now. I have addressed this. Pls take a re-look.
I have manually deprecated the older function keeping the backward compatibility.